### ISSUE TITLE
Changed checksum for Go pkg `operator-sdk@v0.17.1`

### ISSUE THREAD

**Issue by Oblynx** [operator-framework/operator-sdk#3226]
## Bug Report

Checksum for `operator-sdk@v0.17.1` Go package seems to have changed. A project where go.sum mentions the old checksum fails to fetch dependencies since about 1 week:

```
$ go mod vendor
 verifying github.com/operator-framework/operator-sdk@v0.17.1: checksum mismatch
     downloaded: h1:ESV2s2oQsZPQiQ8VfC8S5DzEnO/azXF82Fj++5qpAkw=
     go.sum:     h1:Y/QS7FrEQ0KrURc0NYsbD9n6JhR2B4REs2Ki39RBDno=
 SECURITY ERROR
```

**Comment by estroz** [comment]
Yep, we had to re-release that version because our deploy jobs broken. Use v0.18.1 instead if possible.

We should document this issue in the migration guide and release notes.

/assign

/kind documentation

**Comment by estroz** [comment]
[v0.17.2](https://github.com/operator-framework/operator-sdk/releases/v0.17.2) has been released, and the release notes for v0.17.1 say to not use that version.
